app/services/submit_application.rb

```diff
@@ -53,6 +53,10 @@ def email_address_is_a_bot?(reference)
   def submit_application
     application_choices.each do |application_choice|
       SubmitApplicationChoice.new(application_choice).call
+
+      if application_form.apply_again? && application_form.references_complete?
```

Comment:
Somehow I can't find where `references_complete?` is defined, is this correct? I think we don't need `application_form.apply_again?` here.

Reply:
I think I got this muddled up with the spike version. Things have changed a lot since then and I am pretty sure this piece of logic was moved out of this service. I don't think there is any reason for this PR to add this `if` statement so removing it.

Reply:
I think the reason for the if-statement was that we need to progress the application to `application_complete` if there are already 2 references. Otherwise the application is stuck in `awaiting_references`. One for the follow up that introduces the submission test.